app/main/views.py

```diff
@@ -1,7 +1,59 @@
-from app import app
+import dotenv
+from flask import flash, redirect, render_template, request, url_for
+from flask_login import login_required, current_user
 
+from app.auth.forms import RegistrationForm, LoginForm
+from app.helper import UrlSaver
+from app.main import main
+from app.models import AnonymousUser, User
+from app.main.forms import ShortForm
 
-@app.route('/')
-@app.route('/index')
+
+dotenv.load()
+site_url = dotenv.get('SITE_URL')
+
+
+@main.route('/')
+@main.route('/shorten', methods=['POST'])
+@main.route('/index')
 def index():
-    return "Hello world."
+    if current_user.is_authenticated:
+        return redirect(url_for('main.user_dashboard'))
+    register_form = RegistrationForm()
+    login_form = LoginForm()
+    short_form = ShortForm()
+    short_url = ""
+
+    if short_form.validate_on_submit():
+        user = User.query.filter_by(email='AnonymousUser').first()
+        if not user:
+            user = AnonymousUser.create_anonymous_user()
+
+        short_url = UrlSaver.generate_and_save_urls(
+                            short_form.url.data, user).short_url
+
+        try:
+            if request.headers['Ajax-Test']:
+                return site_url + short_url
+        except KeyError:
+            return render_template('index.html', register_form=register_form,
+                                   login_form=login_form,
+                                   short_form=short_form,
+                                   short_url=site_url + short_url)
+
+    if short_form.errors:
+        flash([value for value in short_form.errors.values()][0][0],
+              'shorten')
+    try:
```

Comment:
Whenever a form fails validation there exist a `.errors` property which is a dict object with the fields that failed validation as keys and a list of error messages as the values. I was trying to get a list of all the error messages from the dict to display to the user. As you would know the `.values()` property of a dictionary returns a dict_values object which does not support indexing, so I had to use a list comprehension to convert it to a python list before using the index to access the error message to flash.